presto-main/src/main/java/io/prestosql/connector/informationschema/InformationSchemaMetadata.java

```diff
@@ -169,8 +164,14 @@ public ConnectorTableHandle getTableHandle(ConnectorSession connectorSession, Sc
         if (!TABLES.containsKey(tableName)) {
             return null;
         }
-
-        return new InformationSchemaTableHandle(catalogName, tableName.getSchemaName(), tableName.getTableName(), defaultPrefixes());
+        if (isTablesEnumeratingTable(tableName)) {
+            Session session = ((FullConnectorSession) connectorSession).getSession();
+            Set<QualifiedTablePrefix> schemaPrefixes = metadata.listSchemaNames(session, catalogName).stream()
```

Comment:
Does it mean we're going to list metadata (eg hit metastore) even when user is doing a local operation like `desc hive.information_schema.tables`?

Reply:
Yes, good point. I moved the listing operation to the page source provider to deal with this.